test/ext/mypy/plugin_files/issue_9156.py

```diff
@@ -0,0 +1,9 @@
+from typing import Type, Any
+
+from sqlalchemy.sql.elements import ColumnElement
+from sqlalchemy.sql.type_api import TypeEngine
+
+col: ColumnElement
+type_: Type[TypeEngine[Any]]
+
+col.cast(type_)
```

Comment:
We could also try with the type instance

Reply:
Done, thanks